src/ninja.cc

```diff
@@ -1118,7 +1118,9 @@ int ReadFlags(int* argc, char*** argv,
   return -1;
 }
 
-int real_main(int argc, char** argv) {
+NORETURN void real_main(int argc, char** argv) {
+  // Use exit() instead of return in this function to avoid potentially
```

Comment:
Couldn't real_main keep using return and main below then does `exit(real_main())`?

Reply:
Nevermind; then we don't prevent destruction of NinjaMain.